python/lsst/ts/m2/model.py

```diff
@@ -0,0 +1,849 @@
+            mirror. The units are the micron and arcsec.
+        temperature : `dict`
+            Temperature of mirror in degree C.
+        axial_forces : `dict`
+            Forces of the axial actuators in Newton.
+        tangent_forces : `dict`
+            Forces of the tangent actuators in Newton.
+        force_balance : `dict`
+            Force balance system contains the information of force and moment.
+            The units are the Newton and Newton * meter.
+        force_balance_system_status : `bool`
+            Force balance system is on or off.
+        lut : `dict`
+            Look-up table (LUT).
+        error_cleared : `bool`
+            Error is cleared or not.
+        """
+
+        # Set the logger
+        if log is None:
+            self.log = logging.getLogger(type(self).__name__)
+        else:
+            self.log = log.getChild(type(self).__name__)
+
+        self.telemetry_interval = telemetry_interval
+
+        self.inclination_source = MTM2.InclinationTelemetrySource.ONBOARD
+        self.zenith_angle = 0.0
+
+        self.n_actuators = 72
+        self.n_tangent_actuators = 6
+
+        self.mirror_position = dict(
+            [(axis, 0.0) for axis in ("x", "y", "z", "xRot", "yRot", "zRot")]
+        )
+
+        self.temperature = self._get_default_temperature()
+
+        self.axial_forces = dict(
+            [
+                (item, np.zeros(self.n_actuators))
+                for item in (
+                    "lutGravity",
+                    "lutTemperature",
+                    "applied",
+                    "measured",
+                    "hardpointCorrection",
+                )
+            ]
+        )
+        self.tangent_forces = dict(
+            [
+                (item, np.zeros(self.n_tangent_actuators))
+                for item in (
+                    "lutGravity",
+                    "lutTemperature",
+                    "applied",
+                    "measured",
+                    "hardpointCorrection",
+                )
+            ]
+        )
+        self.force_balance = dict(
+            [(axis, 0) for axis in ("fx", "fy", "fz", "mx", "my", "mz")]
+        )
+        self.force_balance_system_status = False
+
+        self.lut = dict()
+
+        self.error_cleared = True
+
+    def _get_default_temperature(self, init_temperature=11.0):
+        """Get the default temperature. The unit is degree C.
+
+        Parameters
+        ----------
+        init_temperature : `float`, optional
+            Initial temperature in degree C. (default is 11.0)
+
+        Returns
+        -------
+        `dict`
+            Temperature sensor data. The key is sensor's position or reference.
+        """
+
+        temperature = dict()
+        temperature["ring"] = [init_temperature] * 12
+        temperature["intake"] = [init_temperature] * 2
+        temperature["exhaust"] = [init_temperature] * 2
+        temperature["ref"] = 21.0
+
+        return temperature
```

Comment:
Consider doing this all in one line. Python is slow in creating variables that are not used, because they need to be garbage collected later. Also, consider using lumpy arrays, they are faster then regular python arrays. 

```
return dict(
  ring = np.zeros(12)+init_temperature,
  intake = np.zeros(2)+init_temperature,
  exhaust = np.zeros(2)+init_temperature,
  ref = 21.0.
)
```

Also, maybe make `ref` an optional input with default value 21?

Reply:
This function will only be called for one time, and therefore, maybe the performance is not so important.

Use the `temperature_ref` as an optional input in the commit: 655a34c. Thanks!